src/InvokeCallableCapableTrait.php

```diff
@@ -39,51 +46,71 @@ protected function _invokeCallable($callable, $args)
         $args = $this->_normalizeArray($args);
 
         try {
-            $result = call_user_func_array($callable, $args);
+            $reflection = $this->_createReflectionForCallable($callable);
+            $params     = $reflection->getParameters();
+            $this->_validateParams($args, $params);
         } catch (RootException $e) {
-            throw $this->_createInvocationException(
+            throw $this->_createInvocationException($this->__('Could not invoke callable'), null, $e, $callable, $args);
+        }
+
+        // Invoke the callable
+        try {
+            if ($reflection instanceof ReflectionMethod) {
+                $target = is_object($callable[0])
+                    ? $callable[0]
+                    : null;
+
+                return $reflection->invokeArgs($target, $args);
+            } else {
+                return $reflection->invokeArgs($args);
+            }
+        } catch (RootException $e) {
```

Comment:
The contents of this `try..catch` block almost look like functionality that ought to exist in a separate trait; one that invokes a reflection regardless of whether or not it's a `ReflectionFunction` or `ReflectionMethod`.

What do you think? Perhaps a future improvement, if nothing else.

Reply:
I don't know if this is possible. They are different APIs.

Reply:
True. But we can document `@param ReflectionFunction|ReflectionMethod`, no?

Reply:
True. But what do you invoke it with?

Reply:
Yeah ... ok, disregard for now.